modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -423,14 +436,32 @@ def _compute_axis_labels(self, axis: int, partitions=None):
         -------
         pandas.Index
             Labels for the specified `axis`.
+        tuple of int
+            Size of partitions alongsize specified `axis`.
         """
         if partitions is None:
             partitions = self._partitions
-        new_index, _ = self._partition_mgr_cls.get_indices(axis, partitions)
-        return new_index
+        new_index, internal_idx = self._partition_mgr_cls.get_indices(axis, partitions)
+        return new_index, list(map(len, internal_idx))
+
+    def _filter_empties(self, compute=True):
```

Comment:
```suggestion
    def _filter_empties(self, compute_lengths=True):
```

I think `compute_lengths` is clearer (note you'll have to replace it in several places if you accept this suggestion).